pkg/executor/sortexec/sort.go

```diff
@@ -766,7 +786,11 @@ func (e *SortExec) initCompareFuncs(ctx expression.EvalContext) error {
 func (e *SortExec) buildKeyColumns() {
 	e.keyColumns = make([]int, 0, len(e.ByItems))
 	for _, by := range e.ByItems {
-		col := by.Expr.(*expression.Column)
+		col, ok := by.Expr.(*expression.Column)
```

Comment:
1. what about other expression, for example the function expression?
2. you should add some correctness tests for constant columns in the byItems

Reply:
> 1. what about other expression, for example the function expression?
> 2. you should add some correctness tests for constant columns in the byItems

done